adapter/service/outlvlswitch/event.go

```diff
@@ -0,0 +1,30 @@
+package outlvlswitch
+
+import (
+	"github.com/futurehomeno/cliffhanger/adapter"
+	"github.com/futurehomeno/cliffhanger/event"
+)
+
+type LevelEvent struct {
+	adapter.ServiceEvent
+
+	Level int64
+}
+
+func newLevelEvent(eventType string, hasChanged bool, level int64) *LevelEvent {
+	return &LevelEvent{
+		ServiceEvent: adapter.NewServiceEvent(eventType, hasChanged),
+		Level:        level,
+	}
+}
+
+func WaitForLevelReport() event.Filter {
```

Comment:
If we need a filter like this, we can change the function signature to a generic one (e.g. `WaitFor[T any]()`)

Reply:
@viktor-viktor Please implement generic WaitFor method in event package. Good suggestion.